pkg/filesystem/subvolume.go

```diff
@@ -52,6 +54,78 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	resource := "storageclusters"
+	storageclusters, err := clientsets.ListResourcesDynamically(ctx, "ocs.openshift.io", "v1", resource, clusterNamespace)
+	if err != nil || !strings.Contains(err.Error(), "could not find") {
+		logging.Fatal(err)
+	}
+	if len(storageclusters) == 0 {
+		logging.Info("resource %s was not found on the cluster", resource)
+	}
+
+	for _, storagecluster := range storageclusters {
+		spec, _, err := unstructured.NestedMap(storagecluster.Object, "spec", "externalStorage")
+		if err != nil {
+			fmt.Printf("Error accessing 'externalStorage': %v\n", err)
+			os.Exit(1)
+		}
+		if spec["enable"] != nil {
+			enable = spec["enable"].(bool)
+			return enable
+		}
+	}
+	return enable
+}
+
+// getExternalClusterDetails exports the CEPH_ARGS with required mon-ip, id and key to connect to the
+// ceph cluster.
+func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) (string, string, string) {
+	getsecret := fmt.Sprintf(`kubectl get secret rook-ceph-external-cluster-details -n %s -o=jsonpath='{.data.external_cluster_details}' |  base64 -d`, clusterNamespace)
```

Comment:
i might be missing something, i dont see any reference to `rook-ceph-external-cluster-details` secret in upstream Rook repo. when and who will create this in upstream Rook?

Reply:
This is not in the upstream rook, I was searching for the same, We need to add some check, this is only specific to the external cluster.

Reply:
please use api to fetch secrets